### DISCUSSION TITLE
How to avoid search appending arguments to URL (insiders)

### DISCUSSION THREAD

**Discussion by funkypenguin** [squidfunk/mkdocs-material#2247] [Q&A] [answered]
Hey!

I'm on Insiders, and I've been experimenting with the new sexiness that is search suggestions and highlighting. A common workflow I use is to search my site for the particular page I need, then copy the URL and paste it somewhere else. Since upgrading to Insiders, when I search my URL gets the search term appended as `?h=+<search term>`. Assuming that this was due to highlighting, I disabled search highlighting, but it still happens. 

Here's an example - if you go to https://geek-cookbook.funkypenguin.co.nz and search for "nextcloud", and then click on the first result, the page you end up on is https://geek-cookbook.funkypenguin.co.nz/recipes/nextcloud/?h=+nextcloud

I'm not using any of the fancy search features in Insiders ATM (https://github.com/geek-cookbook/geek-cookbook/blob/c5e313b65c5148173e1e9e23871811581eea6722/mkdocs.yml#L183), but the behaviour is still happening. Is there anything I can do to remove the `?h=<blah>` from the URL after searching?

Thanks!
D

**Comment by squidfunk** [comment, marked as answer]
Sounds like a bug! The URL parameter should only be appended when highlighting is enabled. Could you submit an issue?

**Reply by squidfunk** [reply]
Also, if you have ideas on how to improve the highlight UX, i’d be very interested to discuss.

**Reply by funkypenguin** [reply]
From my perspective, avoiding altering the URL would be ideal. The UX would ideally be that a user could search whatever they wanted to, and the browser URL would simply reflect the page link, for straightforward copy/pasting. If they wanted to share their search parameters, they could use the shared-search function.

**Reply by squidfunk** [reply]
I'm accepting this answer, as it was a bug (that's now been fixed).
